### PR TITLE
fix(tools/operators): Fixed minor typos that came up during testing AND test:added tests for operators WIP

### PR DESCRIPTION
**Pull request by alejandrogranizo**
## Description

This PR introduces the file for testing the operators.py file and fixes minor errors found in edge cases in operators.py methods. Test file is still WIP.

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

## FURTHER NOTES
Closes [eclipse-tractusx/industry-core-hub#54](https://github.com/eclipse-tractusx/industry-core-hub/issues/54)
Closes [#21](https://github.com/eclipse-tractusx/tractusx-sdk/issues/21)

### CODE REVIEW

**Review by matbmoser — CHANGES_REQUESTED** [review]
@mgarciaLKS sonar cloud is not passing, what can we do to fix this?

**Comment by mgarciaLKS** [comment]
@matbmoser @alejandrogranizo It is failing due to the big amount of Duplicated Code in the same class. It is necessary to refactor some methods and variables, in order not to use them more than necessary. I will create a PR to Alejandro's repository improving the tests, and he will merge it.

File: dataspace/tests/tool/test_operators.py [hunk 1]
```diff
@@ -0,0 +1,798 @@
...
+import json
+from tools import op
+
+#------------TEST VARIABLES--------------------------
+
+# print(op.json_string_to_object(""))
+# -----------------TESTS-----------------------------
+
+#Function 1 - json_string_to_object
+def test_json_string_to_object_with_empty_input_should_return_JSONDecodeError():
```
Comment (mgarciaLKS) [inline dataspace/tests/tool/test_operators.py:37]: This class needs to be refactored, 800 lines for one test class is too much, we will work together in improving this class and the tests implemented.

File: dataspace/tests/tool/test_operators.py [hunk 2]
```diff
@@ -0,0 +1,798 @@
...
+import math
+import os
+import pytest
+import json
+from tools import op
+
+#------------TEST VARIABLES--------------------------
+
+# print(op.json_string_to_object(""))
+# -----------------TESTS-----------------------------
```
[the remark this answers is not in the document]
Comment (alejandrogranizo) [reply dataspace/tests/tool/test_operators.py:34]: Yeah, sorry Mathias was still a WIP to not exceed the line limit, and i forgot to delete that line from a while ago, I'll work with @mgarciaLKS to refactor it and take his advice to optimize, sorry about that

**Review by matbmoser — CHANGES_REQUESTED** [review]
I approved because the checks went though and it is not bad, but I don't understand why the file is called V2?

**Comment by alejandrogranizo** [comment]
The V2 file is the file that Mikel sent me to refactor my tests in that structure, I merged it to my fork to combine it all in the original test file but didnt know it updated the PR, ill finish the tests in a single file and send a PR with all in a single file.

**Comment by matbmoser** [comment]
@alejandrogranizo super! thx

**Comment by alejandrogranizo** [comment]
Sorry guys for the chain of commits, I made some suggested changes to operators.py according to test results and method intentions, and some tests needed to get refactored

**Comment by CDiezRodriguez** [comment]
@alejandrogranizo 
If you'd like, you can add these lines to close the issues automatically:
Closes https://github.com/eclipse-tractusx/industry-core-hub/issues/54
Closes https://github.com/eclipse-tractusx/tractusx-sdk/issues/21

**Comment by alejandrogranizo** [comment]
Thanks for the tip Carlos!!
Closes https://github.com/eclipse-tractusx/industry-core-hub/issues/54
Closes https://github.com/eclipse-tractusx/tractusx-sdk/issues/21

**Comment by matbmoser** [comment]
I cant approve until the codes and sonar could things are resolved... Are they false positives?

**Review by matbmoser — APPROVED** [review]
Amazing job @alejandrogranizo! And thx for the detailed review @CDiezRodriguez, @mgarciaLKS !
